aea/decision_maker/base.py

```diff
@@ -301,11 +306,15 @@ def _handle_tx_message(self, tx_message: TransactionMessage) -> None:
         :param tx_message: the transaction message
         :return: None
         """
-        score_diff = self.preferences.get_score_diff_from_transaction(self.ownership_state, tx_message)
-        if score_diff >= 0.0:
-            pass  # TODO: implement tx flow
-            # envelope = Envelope()
-            # self.outbox.put(envelope)
+        amount = 0
+        m_address = _generate_address_from_public_key(self._wallet.public_keys['fetchai'])
```

Comment:
I think I mentioned in a previous review. The funding off the account of an agent has to happen outside of the agent run. Here, you either have money in the account and accept or you don't and reject.

Reply:
I should move the check (if we have money in the wallet and if it is less than the amount we are willing to spend ) to the handler before accepting or declining the proposal. Though I don't want to create a new connection to the ledger to ask for the balance from the handler and that's why I am going to check in the decision-maker for now. Is that ok? (When we have the messaging system complete we can query the decision-maker for that kind of stuff and the decision-maker could pass the balance for example to the handler and then answer ACCEPT or DECLINE)